services/crates/crates-downloads.service.js

```diff
@@ -59,7 +81,25 @@ module.exports = class CratesDownloads extends BaseCratesService {
     }
   }
 
+  _extractDownloads({ variant, json }) {
+    switch (variant) {
+      case 'dv':
+        return json.crate ? json.versions[0].downloads : json.version.downloads
+      case 'dr':
+        return json.crate.recent_downloads
+      default:
+        return json.crate ? json.crate.downloads : json.version.downloads
+    }
+  }
+
   async handle({ variant, crate, version }) {
+    if (variant === 'dr' && version) {
+      /* crates.io doesn't currently expose
+         recent download counts for individual
+         versions */
+      return { message: 'no data' }
+    }
```

Comment:
There's a couple ways we could handle this scenario:

1. Basically the same as what's here, but with our `InvalidParameter` error, perhaps something like:
```js
      throw new InvalidParameter { prettyMessage: 'recent downloads not supported for versions' }
      // Note we'd also need to import
      // const { InvalidParameter } = require('..')
```
2. Just return the recent downloads for crate itself. 

I think I'd prefer option 1, as option 2 may lead to some confusion. Thoughts?

Reply:
I agree that option 2 is confusing! I like your first suggestion better.

Might it also be worth me wrapping the other error (the one that gets returned if the crate version doesn't exist) in a `NotFound` or an `Inaccessible` while I'm at it?

Reply:
> Might it also be worth me wrapping the other error (the one that gets returned if the crate version doesn't exist) in a NotFound or an Inaccessible while I'm at it?

Definitely (and thanks!). `NotFound` would be perfect for that use case 👍

Reply:
Done :)